pkg/internal/local/utils/expect.go

```diff
@@ -0,0 +1,62 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	"context"
+	"errors"
+	"github.com/cenkalti/backoff/v4"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+var DefaultElapsedTime = 30 * time.Second
+
+// WaitUntilSecretCreated - waits until a secret is created in the given namespace with a backoff strategy
+func WaitUntilSecretCreated(ctx context.Context, k8sClient client.Client, name, namespace string) error {
+	b := backoff.NewExponentialBackOff()
+	b.InitialInterval = 5 * time.Second
+	b.MaxElapsedTime = DefaultElapsedTime
+	return backoff.Retry(func() error {
+		Logf("waiting for secret %s to be created...", name)
+		secret := &v1.Secret{}
+		err := k8sClient.Get(ctx, types.NamespacedName{
+			Namespace: namespace,
+			Name:      name,
+		}, secret)
+		if err != nil {
+			return err
+		}
+		return nil
+	}, b)
+}
```

Comment:
What do you think would it make sense to retry the secret creation when it failed via this function ?
https://github.com/cloudoperators/greenhouse/blob/f742b7f7f37915b72f6eca626da3ad071157be76/pkg/retry/retry.go#L49-L70

Reply:
The one time `Job` https://github.com/cloudoperators/greenhouse/blob/main/charts/manager/templates/kube-webhook-certgen.yaml handles certificate injection into webhooks and also the secret creation. This functionality is just waiting for the `Secret` to be created.